src/test/java/de/tum/in/www1/artemis/util/DatabaseUtilService.java

```diff
@@ -1952,6 +1952,78 @@ public Course addCourseWithOneFinishedTextExercise() {
         return course;
     }
 
+    public Course addCourseWithOneFinishedTextExerciseAndSimilarSubmissions(String similarSubmissionText, int studentsAmount) {
+        Course course = ModelFactory.generateCourse(null, pastTimestamp, futureTimestamp, new HashSet<>(), "tumuser", "tutor", "editor", "instructor");
+        addUsers(studentsAmount,1,1,1);
+
+        // Add text exercise to the course
+        TextExercise textExercise = ModelFactory.generateTextExercise(pastTimestamp, futureTimestamp, futureFutureTimestamp, course);
+        textExercise.setTitle("Finished");
+        textExercise.getCategories().add("Text");
+        course.addExercises(textExercise);
+        courseRepo.save(course);
+        exerciseRepo.save(textExercise);
+
+        Set<StudentParticipation> participations = new HashSet<>();
+
+        for (int i = 0; i < studentsAmount; i++){
+            User participant = getUserByLogin("student" + (i + 1));
+            StudentParticipation participation = ModelFactory.generateStudentParticipation(InitializationState.FINISHED, textExercise, participant);
+            participation.setParticipant(participant);
+            TextSubmission submission = ModelFactory.generateTextSubmission(similarSubmissionText, Language.ENGLISH, true);
+            participation = studentParticipationRepo.save(participation);
+            submission.setParticipation(participation);
+            submissionRepository.save(submission);
+
+            participation.setSubmissions(Set.of(submission));
+            participations.add(participation);
+        }
+
+        textExercise.participations(participations);
+        exerciseRepo.save(textExercise);
+
+        return course;
+    }
+
+    public Course addCourseWithOneFinishedModelingExerciseAndSimilarSubmissions(String similarSubmissionModel, int studentsAmount, Course existingCourse) {
```

Comment:
The method name says "addCourse" but if this parameter is set, the name does not conform any more to what the method does. I would suggest you remove this parameter as you have it with the method above. Just use a second course for testing the modeling exercises?

Reply:
I refactored it so that it's more clear formulated now.